inference-engine/src/readers/ir_reader/ie_ir_parser.cpp

```diff
@@ -202,8 +382,54 @@ std::shared_ptr<ICNNNetwork> V10Parser::parse(const pugi::xml_node& root, const
 
     OV_ITT_TASK_NEXT(taskChain, "ConstructCNNNetwork");
 
-    CNNNetwork net(function, _exts);
+    return function;
+}
+
+std::shared_ptr<ICNNNetwork> IRParser::parse(const pugi::xml_node& root, const Blob::CPtr& weights) {
+    return parser->parse(root, weights);
+}
+
+/**
+ * Hold original blob in order to avoid situations when original blob is allocated on stack
+ */
+class WeightsHolderBlob : public TBlob<uint8_t> {
+    Blob::CPtr originBlob;
+
+public:
+    explicit WeightsHolderBlob(const Blob::CPtr& weights) :
+        TBlob<uint8_t>(weights->getTensorDesc(),
+                       weights->cbuffer().as<uint8_t*>()),
+        originBlob(weights) { }
+};
 
+V10Parser::V10Parser(const std::vector<IExtensionPtr>& exts) : _exts(exts) {
+    // Load default opsets
+    opsets["opset1"] = ngraph::get_opset1();
+    opsets["opset2"] = ngraph::get_opset2();
+    opsets["opset3"] = ngraph::get_opset3();
+    opsets["opset4"] = ngraph::get_opset4();
+    opsets["opset5"] = ngraph::get_opset5();
+    opsets["opset6"] = ngraph::get_opset6();
+
+    // Load custom opsets
+    for (const auto& ext : exts) {
+        std::map<std::string, ngraph::OpSet> extOpsets = ext->getOpSets();
+        for (const auto& it : extOpsets) {
```

Comment:
This might reduce some future issues.
```
for (const auto& it : ext->getOpSets()) {
```

I will not bother if `ext->getOpSets()` return temporary.
```
auto && __range = range_expression ;
```
\* from https://en.cppreference.com/w/cpp/language/range-for#Explanation

Reply:
Corrected.